elastic/integrations
# crowdstrike.falcon: Fix Falcon Overview dashboard errors.

## Proposed commit message
`crowdstrike.falcon` data-stream doesn't set `host.hostname` unlike `crowdstrike.alert` or `crowdstrike.host`.
But it uses `host.hostname` in `Falcon Overview` dashboard, which leads to empty data.

Fixes made to `Falcon Overview` dashboard:
- Use `host.name` in `Top Related Hosts` visualisation.
- Use `host.name` field for `Hostname` control.
- Remove unused and unset field control on `observer.address`.
- Add `Severity` name to empty control using the field `crowdstrike.event.SeverityName`.

> [!NOTE]
> `host.name` is used instead of `host.hostname` due to following reasons:
> 1. As per ECS definitions, [`host.name`](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-name) can contain a more generic hostname and applies to several cases, rather than [`host.hostname`](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname) which should contain what `hostname` command returns which may not be always available in the event.
> 2. `host.name` presence is more within Security Solution, such as in `Hosts` flyout.
> 3. `host.name`:[`repo:elastic/detection-rules path:/^rules\// /host\.name/`](https://github.com/search?q=repo%3Aelastic%2Fdetection-rules+path%3A%2F%5Erules%5C%2F%2F+%2Fhost%5C.name%2F&type=code) is also  widely used in Detection Rules than `host.hostname`: [`repo:elastic/detection-rules path:/^rules\// /host\.hostname/`](https://github.com/search?q=repo%3Aelastic%2Fdetection-rules+path%3A%2F%5Erules%5C%2F%2F+%2Fhost%5C.hostname%2F&type=code)



## Checklist


## How to test this PR locally


1. Run system tests with `--defer-cleanup 30m`
`elastic-package stack down && elastic-package build && elastic-package stack up --version=8.13.0 -d -v  && eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v --defer-cleanup 30m --data-streams=falcon`
2. Navigate to local Kibana and check if the `Flacon Overview` dashboard is fixed.
3. Terminate the waiting system test run.
4. Check system tests are successful.
```bash
--- Test results for package: crowdstrike - START ---
╭─────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE     │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├─────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ crowdstrike │ falcon      │ system    │ logfile   │ PASS   │ 36.104079292s │
╰─────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: crowdstrike - END   ---
Done
```


## Related issues


- Closes https://github.com/elastic/integrations/issues/10877
- Closes https://github.com/elastic/integrations/issues/10819

## Screenshots



**Before:**
<img width="1727" alt="Screenshot 2024-11-14 at 3 05 32 PM" src="https://github.com/user-attachments/assets/10b5f68d-605f-4e6f-9039-9c2733fcc429">
<img width="1727" alt="Screenshot 2024-11-14 at 3 06 14 PM" src="https://github.com/user-attachments/assets/af98ae62-49e7-4395-9d6c-5a79440dc61e">

**After:**
<img width="1727" alt="Screenshot 2024-11-14 at 3 23 40 PM" src="https://github.com/user-attachments/assets/442650c7-fe2a-49bb-bb72-8c9d0838c8a9">
